### PR TITLE
[NO CHANGELOG][Add Funds Widget] Create useRoutes

### PR DESCRIPTION
**Pull request by jiyounglee**
Create useRoutes

### CODE REVIEW

**Review by mimi-imtbl — APPROVED** [review]

File: packages/checkout/widgets-lib/src/widgets/add-funds/hooks/useRoutes.ts [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    toAmount: string,
+  ) => {
+    const getAmountData = await Promise.all(
+      getFromAmounts(squid, balances, toChanId, toTokenAddress, toAmount),
+    );
+    const filteredAmountData = getAmountData.filter(
+      (amountData) => amountData !== undefined,
+    ) as AmountData[];
+
+    delay(2000);
```
Comment (jwhardwick) [inline packages/checkout/widgets-lib/src/widgets/add-funds/hooks/useRoutes.ts:119]: Why are we using setTimeout here?

File: packages/checkout/widgets-lib/src/widgets/add-funds/hooks/useRoutes.ts [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+      toAddress,
+      quoteOnly,
+      enableBoost: true,
+    });
+  };
+
+  const getRoutes = async (
+    squid: Squid,
+    balances: TokenBalance[],
+    toChanId: string,
```
Comment (jwhardwick) [inline packages/checkout/widgets-lib/src/widgets/add-funds/hooks/useRoutes.ts:108]: spelling

Comment (jiyounglee) [reply packages/checkout/widgets-lib/src/widgets/add-funds/hooks/useRoutes.ts:108]: We will fix it.

- [ ]

File: packages/checkout/widgets-lib/src/widgets/add-funds/hooks/useRoutes.ts (hunk 1 above)
Comment (jiyounglee) [reply packages/checkout/widgets-lib/src/widgets/add-funds/hooks/useRoutes.ts:119]: We are still spiking for rate limit stuff

File: packages/checkout/widgets-lib/src/widgets/add-funds/hooks/useRoutes.ts [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    ).then((route) => ({
+      amountData: data,
+      route,
+    })).catch((error) => {
+      console.log('error', error);
+      return Promise.resolve(undefined);
+    }));
+
+    const getRouteData = await Promise.all(getRoutePromises);
+    const routesData = getRouteData.filter((routeData) => routeData !== undefined) as RouteData[];
```
Comment (jwhardwick) [inline packages/checkout/widgets-lib/src/widgets/add-funds/hooks/useRoutes.ts:136]: I don't think we need to cast `as RouteData[]` here, TS should take of that

File: packages/checkout/widgets-lib/src/widgets/add-funds/hooks/useRoutes.ts [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    toTokenAddress: string,
+    toAmount: string,
+  ): Promise<AmountData | undefined>[] => {
+    const filteredBalances = balances.filter(
+      (balance) => balance.address !== toTokenAddress,
+    );
+
+    return filteredBalances.map(
+      (balance: TokenBalance): Promise<AmountData | undefined> => {
+        const fromTokenData = squid?.getTokenData(
```
Comment (jwhardwick) [inline packages/checkout/widgets-lib/src/widgets/add-funds/hooks/useRoutes.ts:41]: Don't think we need optional chaining here.

**Pull request merged**